app/routes/companyInterest/components/CompanySemesterGUI.tsx

```diff
@@ -8,128 +7,149 @@ import {
   MultiSelectGroup,
 } from 'app/components/Form';
 import LegoFinalForm from 'app/components/Form/LegoFinalForm';
+import SubmissionError from 'app/components/Form/SubmissionError';
+import { SubmitButton } from 'app/components/Form/SubmitButton';
 import Icon from 'app/components/Icon';
 import Flex from 'app/components/Layout/Flex';
-import type { CompanySemesterEntity } from 'app/reducers/companySemesters';
+import type CompanySemester from 'app/store/models/CompanySemester';
 import { createValidator, required } from 'app/utils/validation';
 import { semesterToText, SemesterNavigation } from '../utils';
 import styles from './CompanyInterest.css';
-import type { FormProps } from 'redux-form';
+import type { FormApi } from 'final-form';
 
 type Props = {
-  push: (arg0: string) => void;
-  events: Array<Record<string, any>>;
-  semesters: Array<CompanySemesterEntity>;
-  autoFocus: any;
-  edit: boolean;
-  editSemester: (arg0: CompanySemesterEntity) => void;
-  addSemester: (arg0: CompanySemesterEntity) => void;
-  activeSemesters: Array<CompanySemesterEntity>;
-} & FormProps;
+  semesters: Array<CompanySemester>;
+  editSemester: (semester: CompanySemester) => Promise<void>;
+  addSemester: (semester: Omit<CompanySemester, 'id'>) => Promise<void>;
+  activeSemesters: Array<CompanySemester>;
+  initialValues: Record<string, any>;
```

Comment:
Is there maybe a point of using a partial type of a more defined `CompanySemester`(`s`) type

Reply:
Yea, you're right. It should be typed properly